packages/common/core-interfaces/api-report/core-interfaces.beta.api.md

```diff
@@ -7,12 +7,29 @@
 // @public
 export type ConfigTypes = string | number | boolean | number[] | string[] | boolean[] | undefined;
 
+// @public
+export function createEmitter<TListeners extends object>(noListeners?: NoListenersCallback<TListeners>): Listenable<TListeners> & IEmitter<TListeners> & HasListeners<TListeners>;
```

Comment:
we don't usually put implementations in core-interface. maybe core-utils would be a better place.

Reply:
I copied over `IEvent` from core-interface as core-utils cannot have core-interface as a dependency.

Reply:
so we usually split types and implementations, types in core-interfaces, and implementations in core-utils, we do this pattern in multiple places. it allows layers to depend on abstractions which have zero bundle impact, and then only get implementation at runtime. i can't really say how effect that pattern has been, and if the gains are truly realized, but it makes sense to align with the pattern regardless, just to keep things consistent.  So i would move the type back to core-interfaces, and only put the implementations (functions, classes) in core-utils.